posts/2022/09-02-dependabot.md

```diff
@@ -0,0 +1,97 @@
+
+After setting it up, we developers can rely on it to open PRs and tirelessly champion the update process with a single-minded purpose no biological lifeform could ever hope to match. That's somewhat useful, in my opinion :)
+
+![A dependabot pull request](dependabot.png "A dependabot pull request")
+
+## What's the catch?
+
+There are 4 requirements. Three are set in stone, and the 4th a little looser.
+
+### 1. GitHub
+
+This one is pretty self-explanatory: you need to be using GitHub. Microsoft is seeking to maximize the value of its product.
+
+### 2. A good test suite
+
+You need a trustworthy test suite that will usually catch dependency regressions.
+
+If you cannot perform minor updates without extensive manual testing, it may be best to prioritize work on the test suite before pursuing Dependabot further. Without "dependable" automated testing, you will only dramatically increase a team's workload as the PR's come in.
+
+### 3. Effective monitoring
+
+It would be best to have good production monitoring to detect issues that the test suite cannot. However, something will slip past once in a while, and you will need that second line of defence. For example, you will want to know if that one endpoint suddenly stops serving requests correctly.
+
+### 4. Trunk-based development (probably)
+
+It would help if you used trunk-based development. Ideally, all updates merged in will end up on production, one by one, as soon as possible. Then, you can revert the last change and avert disaster if something goes wrong. But, of course, if that doesn't happen, you will end up with high-risk deploys containing multiple dependency updates AND business logic changes. And it won't be long before the former gets delayed to benefit the latter, and we end up back where we started.
+
+## Adding Dependabot
+
+Add the following to `.github/dependabot.yml` in your repository.
+
+```yaml
+version: 2
+ updates:
+  - package-ecosystem: "mix"
+  directory: "/"
+  open-pull-requests-limit: 1
+  reviewers:
+  - "thenameofyourteam"
+  schedule:
+  interval: "daily"
+```
+
+It will open one pull request at a time, daily, which is more than enough to keep a repository up to date if addressed daily. Make sure to add your team's name and have it rotate through reviewers.
+
+[Read more about configuring Dependabot](https://docs.github.com/en/code-security/dependabot/dependabot-version-updates/configuring-dependabot-version-updates)
+
+## Handling the PRs as they come in
+
+So Dependabot opens up the first PR. Yay! Let's talk about the next steps.
+
+### 1. Read the changelog (and maybe the git diff)
+
+Dependabot will conveniently attach the changelog to the PR, so read it thoroughly and follow up on any items that seem squirrelly. For example, is the changelog low effort? Is it missing entirely? Then look at the git diff as well.
```

Comment:
```suggestion
Dependabot will conveniently attach the changelog to the PR. Read it thoroughly and follow up on any items that seem squirrelly. For example, is the changelog low effort? Is it missing entirely? Then look at the git diff as well.
```

Reply:
When do I look at the git diff? when something is squirrelly? or in general? what does low effort mean? is that a negative thing? if not, maybe don't list it as an example for squirelly